client_configuration_test.go

```diff
@@ -150,26 +139,18 @@ func TestParseConfigurationFails(t *testing.T) {
 
 			_, err := parseClientConfiguration(fileName)
 
-			if err == nil {
-				t.Fatal("Error should not be nil but was nil")
-			}
+			assertNotNilF(t, err, "parse client configuration error")
 			errMessage := fmt.Sprint(err)
 			expectedPrefix := "parsing client config failed"
-			if !strings.HasPrefix(errMessage, expectedPrefix) {
-				t.Errorf("Error message: \"%s\" should start with prefix: \"%s\"", errMessage, expectedPrefix)
-			}
-			if !strings.Contains(errMessage, tc.expectedErrorMessageToContain) {
-				t.Errorf("Error message: \"%s\" should contain given phrase: \"%s\"", errMessage, tc.expectedErrorMessageToContain)
-			}
+			assertHasPrefixE(t, errMessage, expectedPrefix, "error message")
+			assertStringContainsE(t, errMessage, tc.expectedErrorMessageToContain, "error message")
 		})
 	}
 }
 
 func createFile(t *testing.T, fileName string, fileContents string, directory string) string {
 	fullFileName := path.Join(directory, fileName)
 	err := os.WriteFile(fullFileName, []byte(fileContents), 0644)
```

Comment:
Shouldn't we defer closing file here?

Reply:
I think that no, because we create a directory for the files as `t.TempDir()` and we can read in its description:

"The directory is automatically removed by Cleanup when the test and all its subtests complete."